galaxy/api/v2/tests/test_collection_version_views.py

```diff
@@ -44,6 +50,54 @@ def setUp(self):
             content=self.version, artifact=self.artifact,
             relative_path='mynamespace-mycollection-1.2.3.tar.gz')
 
+    def test_download(self):
+        download_path = \
+            '/download/galaxy/mynamespace-mycollection-1.2.3.tar.gz'
+
+        url = '/api/v2/collections/mynamespace/mycollection' \
+            + '/versions/1.2.3/artifact/'
+
+        # NOTE: This does not follow redirects (follow=True), so response
+        #       will be the 302. Since the redirect is to /download which isn't
+        #       a django url, a get with follow=True will 404 on the second url
+        response = self.client.get(url,
+                                   HTTP_USER_AGENT=mazer_user_agent)
+
+        assert response.status_code == http_codes.HTTP_302_FOUND
+        assert response['Location'] == download_path
+
+    def test_download_count(self):
+        download_count_before = self.collection.download_count
+        log.debug('download_count_before: %s', download_count_before)
+
+        url = '/api/v2/collections/mynamespace/mycollection' \
+            + '/versions/1.2.3/artifact/'
+
+        response = self.client.get(url,
+                                   HTTP_USER_AGENT=mazer_user_agent)
+
+        assert response.status_code == http_codes.HTTP_302_FOUND
+
+        self.collection.refresh_from_db()
+
+        assert self.collection.download_count == download_count_before + 1
+
+    def test_download_count_non_mazer_user_agent(self):
```

Comment:
The extra logging isn't really adding anything, because pytest inspects the frame when an assert fails and reports the values of the variables in the assert expression. So if the assert on the `download_count` vs `download_count_before` fail, it'll report both values.

The logging doesn't hurt, obviously, just an FYI

Reply:
yup, true. Leftovers from test bootstrapping. I'll clean them up.